crates/solver/src/driver.rs

```diff
@@ -528,6 +528,8 @@ impl Driver {
                     }
                 };
                 self.in_flight_orders.mark_settled_orders(block, orders);
+                self.metrics
+                    .transaction_gas_price(receipt.effective_gas_price);
```

Comment:
Alternatively we could track in the solution submission strategies. The difference is that we get a gas price even when the TX fails and we get the gas price earlier. We can update the gas price every time we resubmit. On the other hand for flashbots at least we would sometimes gauge gas prices where the transactions didn't really happen but would have allowed it to happen.

Reply:
I thought about this a bit more.

Really my goal with this metric is to plot the gas estimate and what we actually use in the same plot.

Ideally, if we wanted more metrics, we could do, as you mentioned, pass in metrics to the settlement submission and also record all prices that were used as well as their "result" (like `replaced` if it was replaced because it took too long) and maybe even find a way to incorporate gas price used -> time to mine.

I'll create an issue to address this.

Reply:
https://github.com/gnosis/gp-v2-services/issues/1421